modules/collect/src/main/java/com/opengamma/strata/collect/MapStream.java

```diff
@@ -86,6 +86,17 @@
     return new MapStream<>(multimap.entries().stream());
   }
 
+  /**
+   * Returns a stream of map entries where the keys and values are taken from a collection.
+   *
+   * @param <V>  the key and value type
+   * @param collection  the collection
+   * @return a stream of map entries derived from the values in the collection
+   */
+  public static <V> MapStream<V, V> of(Collection<V> collection) {
+    return of(collection.stream(), key -> key);
+  }
```

Comment:
Might as well add the other option of
`MapStream.of(Stream<V> stream)`
and delegate to that.